Sources/Datadog/RUM/RUMEvent/RUMEventBuilder.swift

```diff
@@ -22,11 +22,17 @@ internal class RUMEventBuilder {
         with model: DM,
         attributes: [String: Encodable]
     ) -> RUMEvent<DM>? {
-        let event = RUMEvent(
-            model: model,
-            attributes: attributes,
-            userInfoAttributes: userInfoProvider.value.extraInfo
-        )
+        var model = model
+
+        if !attributes.isEmpty {
+            model.context = RUMEventAttributes(contextInfo: attributes)
+        }
+
+        if !userInfoProvider.value.extraInfo.isEmpty {
+            model.usr = RUMUser(email: model.usr?.email, id: model.usr?.id, name: model.usr?.name, usrInfo: userInfoProvider.value.extraInfo)
+        }
```

Comment:
With the change we made to `usr`:
```swift
public internal(set) var usr: RUMUser?
```
we can simplify this to:
```swift
model.usr?.usrInf = userInfoProvider.value.extraInfo
```
no?

Reply:
If `model.usr` is nil, then we would loose the extraInfo. Now that makes me wondering if we should get all user Info from the `userInfoProvider`, or event use the `RUMUserInfoProvider` instead?

Reply:
> If model.usr is nil, then we would loose the extraInfo.

Ah, right 👍.

> Now that makes me wondering if we should get all user Info from the `UserInfoProvider`, or event use the `RUMUserInfoProvider` instead?

Indeed, but we can do even better. We're using `UserInfoProvider` here, as this code is older than `RUMUser` added to `RUMDataModels` (through schema). Now, as we use `RUMUserInfoProvider` directly in RUM scopes (e.g. [here in `RUMViewScope`](https://github.com/DataDog/dd-sdk-ios/blob/16a55a14b422fe4cf5dff72fe14a61150d54cf1b/Sources/Datadog/RUM/RUMMonitor/Scopes/RUMViewScope.swift#L309)), the user model is already emitted to all events and there is no need override it here.

However, to remove `userInfoProvider` from here, we'd need to pay off one more piece of technical debt: [the `RUMUserInfoProvider` seems to ignore `usrInfo` attributes](https://github.com/DataDog/dd-sdk-ios/blob/16a55a14b422fe4cf5dff72fe14a61150d54cf1b/Sources/Datadog/RUM/RUMEvent/RUMUserInfoProvider.swift#L17-L27). This seems aligned with the scope of this RUMM-1420 - @maxep do you think this is something we improve now?

Reply:
Ah yes, totally! I will change that!